clis/teliod/src/command_listener.rs

```diff
@@ -59,22 +55,91 @@ impl CommandListener {
                     })?;
                 // wait for a response from telio runner
                 if let Ok(report) = response_rx.await {
-                    // send the response to teliod client
-                    connection
-                        .respond(CommandResponse::StatusReport(report).serialize())
-                        .await?;
+                    return Ok(CommandResponse::StatusReport(report));
+                }
+            }
+            ClientCmd::EnableMeshnet => {
+                info!("Enabling meshnet");
+                let (response_tx, response_rx) = oneshot::channel();
+                #[allow(mpsc_blocking_send)]
+                self.telio_task_tx
+                    .send(TelioTaskCmd::SetMeshnet(true, response_tx))
+                    .await
+                    .map_err(|e| {
+                        error!("Error sending command: {}", e);
+                        TeliodError::CommandFailed(ClientCmd::EnableMeshnet)
+                    })?;
+                // wait for a response from telio runner
+                if let Ok(response) = response_rx.await {
+                    match response {
+                        Ok(()) => {
+                            return Ok(CommandResponse::Ok);
+                        }
+                        Err(e) => {
+                            return Ok(CommandResponse::Err(e.to_string()));
+                        }
+                    }
+                }
```

Comment:
We can replace this daemon response blocks for a function/closure. (including `ClientCmd::GetStatus` case)

Reply:
Good idea, I'll take a look how to make it cleaner.

Reply:
Added `handle_response` helper function to handle responses in a more generic way.